### PR TITLE
Quality of Life Changes + API changes

### PR DESCRIPTION
**Pull request by halotroop2288**
I'll once again be adding onto this whatever I need, but you can merge whenver you want.

- Adds an EntityPreviewControl (***untested***)
- Changes the functionality if the filter method in SlotControl
- Makes some of the nested classes in Test static as they should be to reduce IDE warnings

### CODE REVIEW

**Comment by halotroop2288** [comment]
Ok I'm confused as to why it keeps adding on these same commits even after they've been merged but whatever.

File: src/main/java/io/github/redstoneparadox/oaktree/client/gui/control/EntityPreviewControl.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package io.github.redstoneparadox.oaktree.client.gui.control;
+
+import io.github.redstoneparadox.oaktree.client.gui.ControlGui;
+import net.minecraft.client.gui.screen.ingame.InventoryScreen;
+import net.minecraft.client.util.math.MatrixStack;
+import net.minecraft.entity.LivingEntity;
+
+public class EntityPreviewControl extends Control<EntityPreviewControl> {
```
Comment (RedstoneParadox) [inline src/main/java/io/github/redstoneparadox/oaktree/client/gui/control/EntityPreviewControl.java:11]: Class is missing Javadoc (Yes, I know all of mine don't have it but I'm lazy) and I'm thinking there should be an option to make the entity look at the mouse; I'm fine with that being limited to the player.

File: src/main/java/io/github/redstoneparadox/oaktree/client/gui/control/SlotControl.java [hunk 2]
```diff
@@ -88,13 +88,18 @@ public SlotControl canInsert(@NotNull TriPredicate<ControlGui, SlotControl, Item
...
 	}
-
-	public SlotControl filter(Item... items) {
+	
+	/**
+	 * @param allow Whether to allow the following items to be inserted
+	 * @param items items to allow/disallow
+	 * @return the control itself
+	 */
+	public SlotControl filter(boolean allow, Item... items) {
```
Comment (RedstoneParadox) [inline src/main/java/io/github/redstoneparadox/oaktree/client/gui/control/SlotControl.java:97]: Already did this.

File: src/main/java/io/github/redstoneparadox/oaktree/test/Tests.java [hunk 3]
```diff
@@ -7,8 +7,10 @@
 import io.github.redstoneparadox.oaktree.networking.OakTreeNetworking;
```
Comment (RedstoneParadox) [inline src/main/java/io/github/redstoneparadox/oaktree/test/Tests.java:7]: I'm going to be moving the test classes into their own module before the next release and get everything reorganized, so yeah.

Comment (halotroop2288) [reply src/main/java/io/github/redstoneparadox/oaktree/test/Tests.java:7]: Modules kinda suck. Please reconsider.

Comment (RedstoneParadox) [reply src/main/java/io/github/redstoneparadox/oaktree/test/Tests.java:7]: Sorry, I meant a source set, not a module.

File: src/main/java/io/github/redstoneparadox/oaktree/client/gui/control/SlotControl.java (hunk 2 above)
Comment (halotroop2288) [reply src/main/java/io/github/redstoneparadox/oaktree/client/gui/control/SlotControl.java:97]: Uh... push it so I can rebase.

File: src/main/java/io/github/redstoneparadox/oaktree/client/gui/control/EntityPreviewControl.java (hunk 1 above)
Comment (halotroop2288) [reply src/main/java/io/github/redstoneparadox/oaktree/client/gui/control/EntityPreviewControl.java:11]: Is it good now?

**Comment by halotroop2288** [comment]
God damn it. Why does it insist on doing that?
Definitely squash this...

File: src/main/java/io/github/redstoneparadox/oaktree/client/gui/control/SlotControl.java (hunk 2 above)
Comment (RedstoneParadox) [reply src/main/java/io/github/redstoneparadox/oaktree/client/gui/control/SlotControl.java:97]: You should be able to rebase now.

**Pull request merged**